### ISSUE TITLE
Sigma clipping

### ISSUE THREAD

**Issue by mileslucas** [JuliaAstro/Photometry.jl#13]
This involves implementing some basic statistical functions using standard sigma-clipping
```julia
sigma_clip(data, sigma_low, sigma_high; center=median, std=std)
sigma_clip(data, sigma; kwargs...) = sigma_clip(data, sigma, sigma; kwargs...)
```

`data` is the input data
`sigma` is the number of sigma to clip with
`sigma_low`,`sigma_high` allow asymmetric clipping
`center` is a function for computing the central value (usually mean or median)
`std` is a function for computing the spread (usually std)

The function will return the original data with the clipped values imputed with the appropriate max (ie if the original is greater than `sigma_high * std + mean` then it is replaced with `sigma_high * std + mean`)

**Comment by giordano** [comment]
Note that Julia already has [`clamp`](https://docs.julialang.org/en/v1/base/math/#Base.Math.clamp) and [`clamp!`](https://docs.julialang.org/en/v1/base/math/#Base.Math.clamp!) that do something very similar.  `sigma_clip` would probably be a wrapper around these functions.

**Comment by mileslucas** [comment]
Thanks @giordano that's a great point. 

RE: our discussion about refactoring AstroBase this is a method I could see put there.

**Comment by mileslucas** [comment]
Closed via #14
